src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs

```diff
@@ -568,9 +568,6 @@ private async Task<IActionResult> ExchangePasswordGrantType(OpenIddictRequest re
 
         private async Task<IActionResult> ExchangeAuthorizationCodeOrRefreshTokenGrantType(OpenIddictRequest request)
         {
-            var application = await _applicationManager.FindByClientIdAsync(request.ClientId) ??
-                throw new InvalidOperationException("The application details cannot be found.");
```

Comment:
Not sure abou this change @agriffard 
Why was it being suggested?

Reply:
Because `application` is not used later int he code.

Reply:
Remove the variable then, not the check that it exists. (I'm guessing here, but we wouldn't want to lose the behaviour of checking that the client id exists, I think)
i.e.
```
await _applicationManager.FindByClientIdAsync(request.ClientId) ??
                throw new InvalidOperationException("The application details cannot be found.");
```

Reply:
re-added but it requires to set a variable.

Reply:
Feel free to remove it: OpenIddict itself ensures the application exists and this second call will basically return a cached value anyway. If we don't use it, there's no point doing that (we don't allow disabling OpenIddict's L2-like cache anyway).